CMakeLists.txt

```diff
@@ -24,12 +24,19 @@ find_package(Boost COMPONENTS context fiber REQUIRED)
 
 # Check for CUDA/ROCm and clang
 find_package(CUDA QUIET)
-# We currently search for hipcc to check for ROCm installation
-find_program(HIPCC_COMPILER NAMES hipcc HINTS ${ROCM_PATH})
-set(ROCM_PATH /opt/rocm CACHE PATH "Path to ROCm installation")
-
-if(HIPCC_COMPILER MATCHES "-NOTFOUND")
-  set(ROCM_FOUND false)
+find_package(HIP QUIET)
+
+# In case HIP is not found via the cmake pkgs we fall back to the legacy rocm discovery:
+if (NOT HIP_FOUND)
+  message(STATUS "Could not find hip cmake integration falling back to finding hipcc")
+  find_program(HIPCC_COMPILER NAMES hipcc HINTS ${ROCM_PATH})
+  set(ROCM_PATH /opt/rocm CACHE PATH "Path to ROCm installation")
```

Comment:
`ROCM_PATH` is used later for the `syclcc` configuration, so I think it should be set in any case. Also, if set, I don't think it should be ignored if HIP was found.
Is it possible to use it as a hint for the `find_package()` call?

Reply:
So far the `ROCM_PATH` var was associated with the root directory of the ROCM directory right? My understanding is that `HINTS` can be specified for `find_package` but it should point to the directory containing the hip rocm configuration files, which would be something like: `ROCM_PATH/lib/cmake/` Therefore I don't think that using the `ROCM_PATH` as a hint a good idea, So I think the best is to have both `ROCM_PATH` and `ROCM_PATH`/lib/cmake added to the HINTS, but that is not a perfect solution. At least in my scripts for the build from a spack install I use the `CMAKE_PREFIX_PATH` to specify all the directories necessary for finding HIP and its dependnecies

Reply:
> ROCM_PATH and ROCM_PATH/lib/cmake added to the HINTS, but that is not a perfect solution

What would be the issue? If the hint doesn't work, the user can still override, no?

Reply:
Nothing actually :D